### PR TITLE
Parcel provisioner

### PR DESCRIPTION
**Pull request by ggop**
Some changes as current version doesn't cleanly handle non-CDH parcels and also throws minor errors when more than one parcel is deployed. Changes to the Azure and AWS scripts are identical.

### CODE REVIEW

File: faster-bootstrap-azure/scripts/provisioning/parcel_provisioner.sh [hunk 1]
```diff
@@ -38,8 +38,11 @@ then
   fi
 fi
 
-sudo useradd -r cloudera-scm
-sudo mkdir -p /opt/cloudera/parcels /opt/cloudera/parcel-repo /opt/cloudera/parcel-cache
+# Just in case multiple parcels are being deployed in the one image
+id cloudera-scm 2>&1 >& /dev/null || sudo useradd -r cloudera-scm
+for dir in parcel-repo parcel-cache parcels; do
+  [ -d /opt/cloudera/$dir ] || sudo mkdir -p /opt/cloudera/$dir
```
Comment (mdwn) [inline faster-bootstrap-azure/scripts/provisioning/parcel_provisioner.sh]: Could you explain? http://man7.org/linux/man-pages/man1/mkdir.1.html mkdir -p isn't supposed to fail if the directory exists already. I'm fine with the id change.

File: faster-bootstrap/scripts/provisioning/parcel_provisioner.sh [hunk 2]
```diff
@@ -38,8 +38,11 @@ then
   fi
 fi
 
-sudo useradd -r cloudera-scm
-sudo mkdir -p /opt/cloudera/parcels /opt/cloudera/parcel-repo /opt/cloudera/parcel-cache
+# Just in case multiple parcels are being deployed in the one image
+id cloudera-scm 2>&1 >& /dev/null || sudo useradd -r cloudera-scm
+for dir in parcel-repo parcel-cache parcels; do
+  [ -d /opt/cloudera/$dir ] || sudo mkdir -p /opt/cloudera/$dir
```
Comment (mdwn) [inline faster-bootstrap/scripts/provisioning/parcel_provisioner.sh]: Same question here.

**Comment by mdwn** [comment]
This overall LGTM. @jadair-cloudera  or @bhavanki , could you take a look as well?

Before this is merged, I'd definitely like the commits squashed as well.

**Comment by jadair-cloudera** [comment]
No additional concerns.

**Review by bhavanki — APPROVED** [review]
Aside from Mike's question, looks good to me. Thanks!

File: faster-bootstrap-azure/scripts/provisioning/parcel_provisioner.sh (hunk 1 above)
Comment (ggop) [reply faster-bootstrap-azure/scripts/provisioning/parcel_provisioner.sh]:
```suggestion
id cloudera-scm 2>&1 >& /dev/null || sudo useradd -r cloudera-scm
sudo mkdir -p /opt/cloudera/parcels /opt/cloudera/parcel-repo /opt/cloudera/parcel-cache
```

File: faster-bootstrap/scripts/provisioning/parcel_provisioner.sh (hunk 2 above)
Comment (ggop) [reply faster-bootstrap/scripts/provisioning/parcel_provisioner.sh]:
```suggestion
id cloudera-scm 2>&1 >& /dev/null || sudo useradd -r cloudera-scm
sudo mkdir -p /opt/cloudera/parcels /opt/cloudera/parcel-repo /opt/cloudera/parcel-cache
```

**Comment by ggop** [comment]
Great point @meowfaceman, I've restored  `mkdir -p`.

**Pull request merged**